docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,314 @@
+on Google Cloud Autoscaler (https://cloud.google.com/compute/docs/autoscaler/) and works only
+in Google Compute Engine. It looks on cpu-usage (what is the current load) and pod-cpu-requests
+(what are the cpu request values for pods). It can also scale on memory but for the simplicity
+this document will focus only on the CPU aspect. The user is expected to provide the target
+level of utilization and GCA will adjust the size to meet the target. From the bird’s eye the
+architecture looks as below.
+
+```
+                  +--MIG-----------------------------+
+                  |                                  |
+                  | +--------+ +--------+ +--------+ |
+                  | |        | |        | |        | |
+                  | | Node1  | | Node2  | | NodeX  | |
+                  | |        | |        | |        | +<--+
+                  | +--------+ +--------+ +--------+ |   |
+                  | |cAdvisor| |cAdvisor| |cAdvisor| |   |
+                  | +--------+ +--------+ +----+---+ |   |
+                  |     |          |           |     |   |
+                  +-----|----------|-----------|-----+   |
+                        |          |           |         |
+                        |      +---v----+      |         |
+                        +----> |Heapster| <----+         |
+                               +---+----+                |
+                                   |                     |
+   Kubernetes                      |                     |
+-----------------------------------|---------------------|------
+   Google                          |                     |
+                                   |                     |
+                            +------v-----+       +-------+------+
+                            |Google Cloud|       |MIG Definition|
+                            | Monitoring |       +-------^------+
+                            +------------+               |
+                                   |                     |
+                                   |                     |
+                            +------v-----+               |
+                +------+    |Google Cloud+---------------+
+                |Config|--->| Autoscaler |
+                +------+    +------------+
+```
+
+CAdvisor instances gather metrics for nodes they are running on. Periodically Heapster (a service running
+on one of the nodes) queries all of the available nodes/cAdvisors, aggregates the metrics and pushes the
+usage to Google Cloud Monitoring (Heapster also takes Pod requests from apiserver but this fact is not
+depicted on the diagram). Google Cloud Autoscaler checks the metrics in GCM, compares the values
+and min/max cluster sizes from the config (that is passed through google-specific gcloud tool or web ui,
+and currently set up in kube-up/push) and updates the number of number of nodes in MIG (Managed Instance Group)
```

Comment:
Duplicated `number of` in `the number of number of nodes in MIG`, please remove one.